### PR TITLE
[Merged by Bors] - feat: port the `lift` tactic

### PR DESCRIPTION
**Pull request by urkud**
Mostly works, doesn't simplify if used as `lift a to α with b hb`.

Based on leanprover-community/mathlib@e50b8c2

### CODE REVIEW

**Comment by urkud** [comment]
It already works in some cases but it fails in the test in the file, and I don't understand why.

File: Mathlib/Tactic/Lift.lean [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+#align pi_subtype.can_lift' PiSubtype.canLift'
+
+instance Subtype.canLift {α : Sort _} (p : α → Prop) :
+    CanLift α { x // p x } Subtype.val p where prf a ha :=
+  ⟨⟨a, ha⟩, rfl⟩
+#align subtype.can_lift Subtype.canLift
+
+open Lean in
+/-- If a `Lean.Expr` has form `Lean.Expr.fvar n`, then returns `some n`, otherwise `none`. -/
+def Lean.Expr.fvarId? : Expr → Option FVarId
```
Comment (urkud) [inline Mathlib/Tactic/Lift.lean:65]: Where should this definition go (before it goes to the core)?

**Comment by semorrison** [comment]
@urkud, I don't see any tests in the file, so it's hard to know what help is needed here. It would be good to get this merged as some PRs are waiting on having `lift` now.

**Comment by urkud** [comment]
AFAIK, everything works now. I found and fixed a bug yesterday. I have no time to add tests before the weekend (home alone with 2 kids and end of semester).

**Comment by semorrison** [comment]
Okay! I'm just curious how you know anything works, as it doesn't seem to be run on anything at all.

**Comment by urkud** [comment]
I tried it on some examples but I didn't push any of them.

**Comment by semorrison** [comment]
I was mistaken in thinking that another PR was waiting on this, so there's less hurry. We can add some examples (porting from the mathlib test file), and then do some refactoring.

**Comment by urkud** [comment]
@semorrison It seems that `lift n to nat` doesn't add `0 ≤ n` as a goal.

**Comment by dupuisf** [comment]
This is now ready for review; as far as I can tell it reproduces the behavior of the Lean 3 version.

**Comment by digama0** [comment]
Sorry for the delay, it looks good.

bors merge

**Pull request closed without merging**